src/amo/reducers/versions.js

```diff
@@ -500,42 +500,47 @@ const reducer = (
-      const { collections, shelves } = action.payload;
+      const { homeShelves } = action.payload;
 
       const newVersions = {};
-      for (const shelf of Object.keys(shelves)) {
-        if (shelves[shelf]) {
-          for (const addon of shelves[shelf].results) {
-            if (addon.current_version) {
-              const currentVersion = addon.current_version;
-              let version = createInternalVersion(currentVersion, state.lang);
-              // license and release_notes are omitted from the search endpoint results,
-              // use them from an existing version if available.
-              const { id } = currentVersion;
-              const existingVersion = getVersionById({ id, state });
-              if (existingVersion) {
-                version = {
-                  ...version,
-                  license: existingVersion.license,
-                  releaseNotes: existingVersion.releaseNotes,
-                };
-              }
-              newVersions[id] = version;
-            }
-          }
-        }
-      }
-
-      for (const collection of collections) {
-        if (collection && collection.results) {
-          for (const addon of collection.results) {
-            const version = createInternalVersion(
-              addon.addon.current_version,
-              state.lang,
-            );
-            newVersions[version.id] = version;
-          }
-        }
+      // for (const shelf of Object.keys(shelves)) {
+      //   if (shelves[shelf]) {
+      //     for (const addon of shelves[shelf].results) {
+      //       if (addon.current_version) {
+      //         const currentVersion = addon.current_version;
+      //         let version = createInternalVersion(currentVersion, state.lang);
+      //         // license and release_notes are omitted from the search endpoint results,
+      //         // use them from an existing version if available.
+      //         const { id } = currentVersion;
+      //         const existingVersion = getVersionById({ id, state });
+      //         if (existingVersion) {
+      //           version = {
+      //             ...version,
+      //             license: existingVersion.license,
+      //             releaseNotes: existingVersion.releaseNotes,
+      //           };
+      //         }
+      //         newVersions[id] = version;
+      //       }
+      //     }
+      //   }
+      // }
+
+      // for (const collection of collections) {
+      //   if (collection && collection.results) {
+      //     for (const addon of collection.results) {
+      //       const version = createInternalVersion(
+      //         addon.addon.current_version,
+      //         state.lang,
+      //       );
+      //       newVersions[version.id] = version;
+      //     }
+      //   }
+      // }
+
+      for (const shelves of homeShelves.results) {
+        // eslint-disable-next-line no-console
+        console.log(shelves);
       }
```

Comment:
```suggestion
     for (const shelf of homeShelves.results) {
        for (const addon of shelf.addons) {
          if (addon.current_version) {
            const currentVersion = addon.current_version;
            let version = createInternalVersion(currentVersion, state.lang);
            // license and release_notes are omitted from the search endpoint results,
            // use them from an existing version if available.
            const { id } = currentVersion;
            const existingVersion = getVersionById({ id, state });
            if (existingVersion) {
              version = {
                ...version,
                license: existingVersion.license,
                releaseNotes: existingVersion.releaseNotes,
              };
            }
            newVersions[id] = version;
          }
        }
      }

```

Reply:
I believe this code will work to address the issue you were having in the versions reducer.